protocol/service_v1.go

```diff
@@ -55,6 +66,26 @@ func (s *serviceV1) Notify(ctx context.Context, req *proto.NotifyReq) (*empty.Em
 	return &empty.Empty{}, nil
 }
 
+// verifyNotifyRequest verifies the notify request to the peer
+func verifyNotifyRequest(request *proto.NotifyReq) error {
+	// Make sure the request is formed
+	if request == nil {
+		return errMalformedNotifyRequest
+	}
```

Comment:
Is there possibility of that request is nil?

Reply:
This is actually a good question 👀 

I believe it can in fact be `nil`, since it's a param that the peer passes in (`req` can be potentially `nil`):
https://github.com/0xPolygon/polygon-edge/blob/f1566f2c0c4529b4aff85c3756a9d240ed8a045a/protocol/syncer.go#L338